utils/stall_free.hh

```diff
@@ -203,19 +203,17 @@ future<> clear_gently(T&&... o) {
 }
 
 template <SharedPointer T>
-future<> clear_gently(T& o) noexcept {
+future<> clear_gently(T& ptr) noexcept {
+    auto o = std::exchange(ptr, nullptr);
     if (o.use_count() == 1) {
-        return internal::clear_gently(const_cast<std::remove_const_t<typename T::element_type>&>(*o));
+        co_await internal::clear_gently(const_cast<std::remove_const_t<typename T::element_type>&>(*o));
     }
-    return make_ready_future<>();
 }
 
 template <SmartPointer T>
-future<> clear_gently(T& o) noexcept {
-    if (auto p = o.get()) {
-        return internal::clear_gently(const_cast<std::remove_const_t<typename T::element_type>&>(*p));
-    } else {
-        return make_ready_future<>();
+future<> clear_gently(T& ptr) noexcept {
+    if (auto o = std::exchange(ptr, nullptr)) {
+        co_await internal::clear_gently(const_cast<std::remove_const_t<typename T::element_type>&>(*o));
```

Comment:
Why the coroutinization?

Reply:
converted to continuations in [71b19ac](https://github.com/scylladb/scylladb/commit/71b19acc30f32721a11d4e2adc24ef4b6c478cda).

Regarding exceptions, I think it's a good requirement for clear_gently to never fail.
It reminds me of https://github.com/scylladb/scylla-enterprise/issues/5580#issuecomment-3218065247
where ~small_vector hit std::bad_alloc in [std::destroy](https://en.cppreference.com/w/cpp/memory/destroy.html) which shockingly isn't noexcept and it failing on std::bad_alloc is really ironic..